gadgets/trace_exec/test/trace_exec_test.go

```diff
@@ -98,75 +101,97 @@ func TestTraceExec(t *testing.T) {
-					Uid:         1000,
-					Gid:         1111,
-					UpperLayer:  false,
-					PupperLayer: true,
-
-					// Check the existence of the following fields
-					MountNsID: utils.NormalizedInt,
-					Timestamp: utils.NormalizedStr,
-					Pid:       utils.NormalizedInt,
-					Ppid:      utils.NormalizedInt,
-					Loginuid:  utils.NormalizedInt,
-					Sessionid: utils.NormalizedInt,
-				},
-			}
-
-			normalize := func(e *traceExecEvent) {
-				utils.NormalizeCommonData(&e.CommonData)
-				utils.NormalizeString(&e.Runtime.ContainerID)
-				utils.NormalizeInt(&e.MountNsID)
-				utils.NormalizeString(&e.Timestamp)
-				utils.NormalizeInt(&e.Pid)
-				utils.NormalizeInt(&e.Ppid)
-				utils.NormalizeInt(&e.Loginuid)
-				utils.NormalizeInt(&e.Sessionid)
-			}
-
-			match.MatchEntries(t, match.JSONMultiObjectMode, output, normalize, expectedEntries...)
-		},
-	))
+	runnerOpts = append(runnerOpts,
+		igrunner.WithFlags("--paths"),
+		igrunner.WithValidateOutput(
+			func(t *testing.T, output string) {
+				expectedEntries := []*traceExecEvent{
+					// outer sh
+					{
+						CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+						Comm:       "sh",
+						Cwd:        "/",
+						Args:       strings.Join(shArgs, " "),
+						UpperLayer: false,
+
+						// Check the existence of the following fields
+						MountNsID: utils.NormalizedInt,
+						Timestamp: utils.NormalizedStr,
+						Pid:       utils.NormalizedInt,
+						Ppid:      utils.NormalizedInt,
+						Loginuid:  utils.NormalizedInt,
+						Sessionid: utils.NormalizedInt,
+					},
+					// inner sh
+					{
+						CommonData: utils.BuildCommonData(containerName, commonDataOpts...),
+						Comm:       "sh",
+						Cwd:        "/",
+						Args:       strings.Join(innerShArgs, " "),
+						Uid:        1000,
+						Gid:        1111,
+						UpperLayer: true,
+
+						// Check the existence of the following fields
+						MountNsID: utils.NormalizedInt,
+						Timestamp: utils.NormalizedStr,
+						Pid:       utils.NormalizedInt,
+						Ppid:      utils.NormalizedInt,
+						Loginuid:  utils.NormalizedInt,
+						Sessionid: utils.NormalizedInt,
+					},
+					// sleep
+					{
+						CommonData:  utils.BuildCommonData(containerName, commonDataOpts...),
+						Comm:        "sleep",
+						Cwd:         "/",
+						Args:        strings.Join(sleepArgs, " "),
+						Uid:         1000,
+						Gid:         1111,
+						UpperLayer:  false,
+						PupperLayer: true,
+
+						// Check the existence of the following fields
+						MountNsID: utils.NormalizedInt,
+						Timestamp: utils.NormalizedStr,
+						Pid:       utils.NormalizedInt,
+						Ppid:      utils.NormalizedInt,
+						Loginuid:  utils.NormalizedInt,
+						Sessionid: utils.NormalizedInt,
+					},
+					// test cwd
+					{
+						CommonData:  utils.BuildCommonData(cwdTestContainerName, commonDataOpts...),
+						Comm:        "sleep",
+						Cwd:         "/tmp",
+						Args:        "/bin/sleep 1",
```

Comment:
```suggestion
						Args:        strings.Join(sleepArgs, " "),
```

BTW, we use `sleepArgs` always with `strings.Join`. Probably, we should define it already as a string `"/bin/sleep 1"`.

Reply:
The thing here is that `Args` should be a string array instead of a string (see #3032), we took this test for the builtin one, so using strings.Joins was a quick way to adapt the test.